### PR TITLE
Pull Request for Feature validations and model specs

### PR DESCRIPTION
**Pull request by chuaindia**
In this PR following activities are incorporated : 

- Following validations for the User model are added ---
         The Name must not be blank.
         PostsCounter must be an integer greater than or equal to zero.
- Following validations for the Post model are added ---
         The title must not be blank.
         The title must not exceed 250 characters.
         CommentCounter must be an integer greater than or equal to zero.
         LikeCounter must be an integer greater than or equal to zero.
- Unit specs for all models' methods and validations are added.

### CODE REVIEW

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @chuaindia,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] Most of your tests are failing. Kindly fix them.

  image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  subject { User.new(Name: 'Tom', Image: 'https://unsplash.com/photos/F_-0BxGuVvo', Bio: 'Teacher from Mexico.') }
+
+  before { subject.save }
+
+  before(:each) do
+    @post = Post.create(AuthorId: subject.id, Title: 'dummy post', Text: 'This is a test for the latest_posts method')
```
Comment (KayLemba) [inline spec/models/comment_spec.rb]: As the previous code reviewer suggested kindly fix your tests. Whenever I try running your tests Im getting the error below.

test

**Review by KayLemba — CHANGES_REQUESTED** [review]
Hi @chuaindia   👋
Good job so far working on the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight 🎉
No linter errors. ✔️
GitHub flow is followed. ✔️

Required Changes ♻️
Check the comments under the review in the inline comments.
 
Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    expect(user).to_not be_valid
+  end
+
+  it 'Name should be present' do
+    user = User.create(Name: 'Tanusri', Image: 'https://unsplash.com/photos/F_-0BxGuVvo', Bio: 'Microverse')
+    user.Name = nil
+    expect(user).to_not be_valid
+  end
+
+  it 'Has a latest_posts method' do
```
Comment (Ismailco) [inline spec/models/user_spec.rb]: - [ ] As the previous reviewer mentioned kindly make sure to fix these errors so the rspec tests are passing 👇🏼 
Screenshot_20230128_204722

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  it 'Title should not exceed 250 characters' do
+    subject.Title = 'a' * 256
+    expect(subject).to_not be_valid
+  end
+
+  it 'Has a latest_comments method' do
+    post = Post.new(Title: 'one post', CommentCounter: 5, LikeCounter: 8)
+    expect(post).to respond_to(:latest_comments)
+  end
```
Comment (Ismailco) [inline spec/models/post_spec.rb]: - [ ] As the previous reviewer mentioned kindly make sure to fix these errors so the rspec tests are passing 👇🏼 
Screenshot_20230128_204700

File: spec/models/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                              Text: 'Unit test for the method of most_recent_comments')
+  end
+
+  it 'Correct AuthorId is present' do
+    like = Like.new(AuthorId: subject.id, PostId: @post.id)
+
+    id = like.AuthorId
+
+    expect(id).to eq(subject.id)
+  end
```
Comment (Ismailco) [inline spec/models/like_spec.rb]: - [ ] As the previous reviewer mentioned kindly make sure to fix these errors so the rspec tests are passing 👇🏼 
Screenshot_20230128_204641

File: spec/models/comment_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    @post = Post.create(AuthorId: subject.id, Title: 'check post',
+                        Text: 'Unit test for the method of most_recent_posts')
+    @comment = Comment.create(AuthorId: subject.id, PostId: @post.id,
+                              Text: 'Unit test for the method of most_recent_comments')
+  end
+
+  it 'Text should match with the input text' do
+    text = @comment.Text
+
+    expect(text).to eq('Unit test for the method of most_recent_comments')
```
Comment (Ismailco) [inline spec/models/comment_spec.rb]: - [ ] As the previous reviewer mentioned kindly make sure to fix these errors so the rspec tests are passing 👇🏼 
Screenshot_20230128_204624

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @chuaindia, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  it 'should test for integers greater than 0' do
+    subject.CommentCounter = 11
+    expect(subject.comments_counter).to eql 11
+  end
+
+  it 'should respond to recent_five_comments method' do
+    expect(subject).to respond_to(:latest_comments)
+  end
```
Comment (Nemwel-Boniface) [inline spec/models/post_spec.rb:32]: - [ ] Unfortunately, this tests still fail on my end despite you making the changes requested by the previous reviewers. Kindly look into why this is the case

image

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  it 'should test for integers greater than 0' do
+    subject.PostCounter = 11
+    expect(subject).to be_valid
+  end
+
+  it 'should respond to recent_three_posts method' do
+    expect(subject).to respond_to(:latest_posts)
+  end
```
Comment (Nemwel-Boniface) [inline spec/models/user_spec.rb:35]: - [ ] Unfortunately, this tests still fail on my end despite you making the changes requested by the previous reviewers. Kindly look into why this is the case

image

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  it 'should test for integer' do
+    subject.PostCounter = 'a'
+    expect(subject).to_not be_valid
+  end
+
+  it 'should test for integers lesser than 0' do
+    subject.PostCounter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (Nemwel-Boniface) [inline spec/models/user_spec.rb:26]: - [ ] Unfortunately, this tests still fail on my end despite you making the changes requested by the previous reviewers. Kindly look into why this is the case

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @chuaindia    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

I tried reaching out to you via slack but unfortunately, you were not there.

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by chuaindia** [comment]
I don't understand this part ..... I  am unable to come out of this error
.... Probably because of my device ..... The active load error or something
.... I am really faded up regarding the db:migrate ...... Probably the
postgresql is not working ..... I googled a lot but problem can not be
solved .....
Feeling really fustrated that how this error prevails how I can continue my
work ,....
Tanusri

**Pull request merged**